### PR TITLE
#2437 adding action in BOY to open resources in phoebus

### PR DESCRIPTION
**Pull request by shroffk**
There is a configurable extra action which would allow users to launch resources using phoebus applications

### CODE REVIEW

**Comment by shroffk** [comment]
image

**Comment by shroffk** [comment]
@berryma4 how are we to manage PR's

**Comment by berryma4** [comment]
Yes, the Travis build goes over the time limit. I just run the build.

I'm fine with the PR if Will is too.

**Comment by shroffk** [comment]
@willrogers Eric is waiting for your input

File: applications/opibuilder/opibuilder-plugins/org.csstudio.opibuilder/src/org/csstudio/opibuilder/preferences/PreferencesHelper.java [hunk 1]
```diff
@@ -501,4 +504,28 @@ public static boolean isAboutShowLinks(){
...
+     */
+    public static String[] supportedResourcesPhoebus(){
+          String resources = getString(PHOEBUS_RESOURCES);
+
+          if(resources == null || resources.trim().isEmpty())
+              resources = "bob";
+          try {
+              return StringSplitter.splitIgnoreInQuotes(resources, ITEM_SEPARATOR, true);
+          } catch (Exception e) {
+              ErrorHandlerUtil.handleError("Failed to parse hidden_widgets preference", e);
```
Comment (willrogers) [inline applications/opibuilder/opibuilder-plugins/org.csstudio.opibuilder/src/org/csstudio/opibuilder/preferences/PreferencesHelper.java]: `hidden_widgets`?

File: applications/opibuilder/opibuilder-plugins/org.csstudio.opibuilder/preferences.ini [hunk 2]
```diff
@@ -120,3 +120,8 @@ mobile_startup_opi=/BOY Examples/main.opi
 
 # Show links section on 'About WebOPI...' popup in RAP version
 about_show_links=true
+
+# use phoebus integration
+use_phoebus=true
```
Comment (willrogers) [inline applications/opibuilder/opibuilder-plugins/org.csstudio.opibuilder/preferences.ini]: Better to default to false?

**Comment by willrogers** [comment]
I made a couple of comments, but it generally looks fine.

File: applications/opibuilder/opibuilder-plugins/org.csstudio.opibuilder/preferences.ini (hunk 2 above)
Comment (berryma4) [reply applications/opibuilder/opibuilder-plugins/org.csstudio.opibuilder/preferences.ini]: I agree, false makes more sense

File: applications/opibuilder/opibuilder-plugins/org.csstudio.opibuilder/preferences.ini [hunk 3]
```diff
@@ -119,4 +119,9 @@ mobile_startup_opi=/BOY Examples/main.opi
 
 
 # Show links section on 'About WebOPI...' popup in RAP version
-about_show_links=true
+about_show_links=false
+
+# use phoebus integration
+use_phoebus=true
```
Comment (berryma4) [inline applications/opibuilder/opibuilder-plugins/org.csstudio.opibuilder/preferences.ini]: This is still true

**Comment by berryma4** [comment]
@shroffk default use_phoebus is still true, could you change it to false

**Comment by berryma4** [comment]
I think you change the wrong line:
https://github.com/ControlSystemStudio/cs-studio/commit/e08eb5ad5bfaaea3fdba2e487671ebbdab7e75e6#diff-bc49ee5bf2f29615acd90e58617543ccL122

**Pull request merged**